src/server/handlers/manager/topdown_msgs.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+//! List top down messages
+
+use std::str::FromStr;
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use fvm_shared::clock::ChainEpoch;
+use ipc_gateway::CrossMsg;
+use ipc_sdk::subnet_id::SubnetID;
+use serde::{Deserialize, Serialize};
+
+use crate::server::handlers::manager::check_subnet;
+use crate::server::handlers::manager::subnet::SubnetManagerPool;
+use crate::server::JsonRPCRequestHandler;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TopDownMsgsParams {
+    pub subnet_id: String,
+    pub start_epoch: ChainEpoch,
+    pub end_epoch: ChainEpoch,
+}
+
+/// The epoch of the latest top-down checkpoint executed
+pub(crate) struct TopDownMsgsHandler {
+    pool: Arc<SubnetManagerPool>,
+}
+
+impl TopDownMsgsHandler {
+    pub(crate) fn new(pool: Arc<SubnetManagerPool>) -> Self {
+        Self { pool }
+    }
+}
+
+#[async_trait]
+impl JsonRPCRequestHandler for TopDownMsgsHandler {
```

Comment:
I think the current `sdk` is more or less doing this? One can just pass the url and token to it and it will point to the correct `ipc-agent` if that's what you mean.